File: HotelWeb/src/app/models/soba.ts
import { SlikaSobe, UpsertNaslovnaSlikaSobe } from './slikaSobe';

export interface Soba {
  id: string;
  naziv: string;
  brojKreveta: number;
  cijenaNocenja: number;
  urlSlike: string;
  hotelNaziv: string;
}
export interface UpsertSoba {
  naziv: string;
  brojKreveta: number;
  cijenaNocenja: number;
  naslovnaSlika: UpsertNaslovnaSlikaSobe | null;
  hotelId: string;
  slike?: SlikaSobe[];
}

export interface SobaFilter {
  brojKreveta?: number;
  cijenaNocenjaMin?: number;
  cijenaNocenjaMax?: number;
  hotelId?: string;
}

File: HotelAPI/Models/Entities/Soba.cs
namespace HotelAPI.Models.Entities
{
    public class Soba : Entity
    {
        public required string Naziv { get; set; }
        public int BrojKreveta { get; set; }
        public int CijenaNocenja { get; set; }
        public Guid HotelId { get; set; }
        public Hotel? Hotel { get; set; }
        public string? UrlSlike { get; set; }
        public ICollection<SlikaSobe>? SlikeSobe { get; set; }
        public ICollection<Rezervacija>? Rezervacije { get; set; }
        public bool IsDeleted { get; set; } = false;

    }
}
    